distributed_shampoo/fsdp_shampoo.py

```diff
@@ -522,24 +601,20 @@ def _init_group(
                     "Sparse parameters are not currently supported by Shampoo."
                 )
 
-            # Initialize momentum and exponential moving average of gradient.
+            # Initialize momentum of gradient.
             if momentum_param != 0.0 and MOMENTUM not in state:
                 state[MOMENTUM] = torch.zeros_like(
                     p.grad, memory_format=torch.preserve_format
                 )
-            if beta1 != 0.0 and EXP_AVG not in state:
-                state[EXP_AVG] = torch.zeros_like(
-                    p.grad, memory_format=torch.preserve_format
-                )
 
             # Generate split lists.
-            split_params.extend(
-                state[PRECONDITIONERS].apply_split(p, return_split_blocks=True)
-            )
-            split_momentum_directions.extend(
-                state[PRECONDITIONERS].apply_split(
-                    state[MOMENTUM], return_split_blocks=True
+            split_params.extend(state[PRECONDITIONERS].apply_split(p))
+            if grad_is_preconditioned:
+                split_preconditioned_grads.extend(
+                    state[PRECONDITIONERS].retrieve_preconditioned_grad()
                 )
```

Comment:
Yes, this function should be used for both `CommunicationShampooPreconditioner` and `SplitShampooPreconditioner`. I am essentially using the flag `grad_is_preconditioned` to indicate which convex shape recovery method is being used (the communication one should retrieve the preconditioned gradients, while the splitting one should just leave the list empty). I agree that maybe we can do away with this function and just incorporate it into the step function.

Reply:
I see. My point is that we should be careful of this - if we don't call this function with `grad_is_preconditioned` set correctly, we could run into the danger of calling a function in `state[PRECONDITIONERS]` that doesn't exist.

Reply:
I see what you mean. My original thought was that maybe in the future we'd like to store preconditioned grad in buffers for all methods, in which case `grad_is_preconditioned` would not necessarily have a one-to-one correspondence to the convex shape recovery method. But in the current case it would definitely be safer to check for `ConvexShapeRecoveryMethod` than using the separate boolean.